Scheduled Updates: Pass status fields directly

obenland opened the pull request:
I don't think we need to register rest field to add status information to the schedule response.

This PR started by me moving the callback to the `Scheduled_Updates` class, but the more I thought about it the less sense it made to keep it separate. I also copied over the `prepare_response_for_collection()` from Core's Plugins endpoint—not sure how good of an idea that is. Just found it odd that we'd create a `WP_REST_Response` object and then pull the `data` property? Don't know.

For the healthcheck paths that we'll add in i2 we'll have to think about something similar.

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Remove register_rest_field callback
* Add status information to the response data directly.

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->
No

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

Unit tests should cover it.

obenland committed:
Scheduled Updates: Pass status fields directly

ouikhuan reviewed (dismissed):
Tested with both unit test and on my testing WoA site. Works as expected!

> I also copied over the prepare_response_for_collection() from Core's Plugins endpoint—not sure how good of an idea that is. Just found it odd that we'd create a WP_REST_Response object and then pull the data property? Don't know.

Although I didn't see any usage of prepare_response_for_collection in the Jetpack repo, but it does shows up a lot in the wpcom database. By doing some search around this seems to be a good practice for REST API.

zaerl commented:
My 2 cents.

I'm not 100% confident this is the best of the changes we have introduced here. In another PR, we added the calls to `register_rest_field` because it's the standard way a rest field is added. https://github.com/Automattic/jetpack/pull/36277#discussion_r1525413269

In our endpoint, we needed to add something to some data **we** created, not some custom data to something output by core functions (that's what `register_rest_field` has been made), so it is okay.

obenland commented:
> I'm not 100% confident this is the best of the changes we have introduced here. In another PR, we added the calls to register_rest_field because it's the standard way a rest field is added. https://github.com/Automattic/jetpack/pull/36277#discussion_r1525413269

Agreed, hence this PR. I think introducing the `prepare_item_for_response()` method was still the right thing to do.
I guess it all depends on what we want the update-schedules endpoint to look like etc.

zaerl reviewed (approved):
It's ok for me. Good work :shipit:
